skopt/optimizer/optimizer.py

```diff
@@ -195,6 +196,7 @@ def tell(self, x, y, fit=True):
             only be fitted after `n_random_starts` points have been queried
             irrespective of the value of `fit`.
         """
+        is_categorical = all([isinstance(dim, Categorical) for dim in self.space.dimensions])
```

Comment:
I need to check once that all are dimensions are Categorical before instantiating the space, no?

The reason being that if all are Categorical, I need to use a different transform.